tests/test_datasets.py

```diff
@@ -107,19 +127,44 @@ def test_register_config(manager, broker):
     assert config_dump['hash'] == manager._make_hash(expected_config_dump)
 
 
-# TODO: register stuff here, then with a new broke test recovery in test_recover
+# TODO: register stuff here, then with a new broker test recovery in test_recover
 def test_register(manager, broker):
     pass
 
 
-def test_recover(manager, broker):
-    state_id = manager.register_state({'foo': "bar"}, "test")
+def test_recover(manager, broker, simple_ds):
+    dset_id = simple_ds[0]
 
-    dset_id = manager.register_dataset(state_id, None, ["test"], True)
+    # Give archiver a moment and make broker release dump file by registering another state.
+    time.sleep(2)
+    manager.register_config({'blubb': 1})
+    time.sleep(.1)
 
     ds = manager.get_dataset(dset_id)
     state = manager.get_state("test")
     assert state == {"foo": "bar", "type": "test"}
     assert ds["is_root"] is True
     # TODO: fix hash function # assert ds["state"] == manager._make_hash(state)
     assert ds["types"] == ["test"]
+
+
+def test_archiver(archiver, simple_ds, manager):
+    dset_id = simple_ds[0]
+    state_id = simple_ds[1]
+
+    # Open database connection
+    chimedb.connect()
```

Comment:
Be careful with this. Because of the fact you've done this via a `.chimedbrc` file, if anyone runs the tests from outside this directory (i.e. does `pytest tests/` from the root dir) it may actually insert things into the real chimedb.

Reply:
good catch. fixed by setting the environment variable that forces chimedb to use the .chimedbrc file